diff --git a/build/media-suite_compile.sh b/build/media-suite_compile.sh
index 41855e4a..c17975fe 100644
--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -2375,37 +2375,17 @@ if [[ $ffmpeg != no ]]; then
     enabled libmodplug && do_addOption --extra-cflags=-DMODPLUG_STATIC && do_pacman_install libmodplug
     enabled libopenjpeg && do_pacman_install openjpeg2
     if enabled libopenh264; then
-        # We use msys2's package for the header and import library so we don't build it, for licensing reasons
-        do_pacman_install openh264
-        if [[ -f $MINGW_PREFIX/lib/libopenh264.dll.a.dyn ]]; then
-            # backup the static library
-            mv -f "$MINGW_PREFIX"/lib/libopenh264.a{,.bak}
-            # use the import library as a phony static library, as mpv doesn't look for .dll.a
-            mv -f "$MINGW_PREFIX"/lib/libopenh264.{dll.a.dyn,a}
-        fi
-        [[ -f $MINGW_PREFIX/lib/libopenh264.dll.a ]] && mv -f "$MINGW_PREFIX"/lib/libopenh264.{dll.,}a
-        _openh264_ver=2.6.0
-        _pacman_openh264_ver=$(pacman -Q "${MINGW_PACKAGE_PREFIX}-openh264" | awk '{print $2}')
-        if [[ $(vercmp.exe $_openh264_ver "$_pacman_openh264_ver") -ne 0 ]]; then
-            do_simple_print "${orange}Openh264 version differs from msys2's, current: $_openh264_ver, msys2: $_pacman_openh264_ver${reset}"
-            do_simple_print "${orange}Check if this is the latest suite and update if possible, else open an issue${reset}"
-        fi
-        if test_newer "$MINGW_PREFIX"/lib/libopenh264.dll.a "$LOCALDESTDIR/bin-video/libopenh264-7.dll" ||
-            ! get_dll_version "$LOCALDESTDIR/bin-video/libopenh264-7.dll" | grep -q "$_openh264_ver"; then
-            pushd "$LOCALDESTDIR/bin-video" >/dev/null || do_exit_prompt "Did you delete the bin-video folder?"
-            if [[ $bits = 64bit ]]; then
-                _sha256=dab5f2a872777f9a58b69bfa9fbcf20d9f82f2d6ec91383fd70bff49bd34ac9f
-            else
-                _sha256=a9445ed1fa2ce9665b22461a7ed0aeb52274add899aa55a93ef6278dbc17c90d
-            fi
-            do_wget -c -r -q -h $_sha256 \
-            "http://ciscobinary.openh264.org/openh264-${_openh264_ver}-win${bits%bit}.dll.bz2" \
-                libopenh264.dll.bz2
-            [[ -f libopenh264.dll.bz2 ]] && bunzip2 -f libopenh264.dll.bz2
-            mv -f libopenh264.dll libopenh264-7.dll
-            popd >/dev/null || do_exit_prompt "Did you delete the previous folder?"
-        fi
-        unset _sha256 _openh264_ver
+
+		_check=(libopenh264.a openh264.pc wels/codec_api.h)
+		if do_vcs "$SOURCE_REPO_LIBOPENH264"; then
+			do_uninstall "${_check[@]}"
+
+			extracommands=("-Dtests=disabled")
+			extracommands+=(--default-library=static)
+			do_mesoninstall "${extracommands[@]}"
+			do_checkIfExist
+			unset extracommands
+		fi
     fi
     enabled chromaprint && do_addOption --extra-cflags=-DCHROMAPRINT_NODLL &&
         { do_pacman_remove fftw; do_pacman_install chromaprint; }
